src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -218,16 +218,19 @@ def get_best_available_genome_id(
         str | None: ID for the genome, if present, otherwise None.
     """
 
+    best_id = None
     if 'RefSeq_accession' in genome_id_data:
-        return genome_id_data['RefSeq_accession']
+        best_id = genome_id_data['RefSeq_accession']
     if 'GenBank_accession' in genome_id_data:
-        return genome_id_data['GenBank_accession']
+        best_id = genome_id_data['GenBank_accession']
     if 'JGI_Genome_ID' in genome_id_data:
-        return genome_id_data['JGI_Genome_ID']
+        best_id = genome_id_data['JGI_Genome_ID']
```

Comment:
This way `best_id` is overwritten within each if; for example, if `'RefSeq_accession'` is present, `best_id` takes its value, but then if `'GenBank_accession'` is present as well `best_id` is overwritten. Either we reverse the order of the if statements, or we go back to the previous logic and handle the empty ids differently.

Reply:
Good catch! Will be fixed.